ui/tests/acceptance/access/identity/_shared-tests.js

```diff
@@ -1,71 +1,70 @@
-  });
-  indexPage.delete().confirmDelete();
+  await showPage.flashMessage.waitForFlash();
+  assert.ok(
+    showPage.flashMessage.latestMessage.startsWith('Successfully saved', `${itemType}: shows a flash message`)
+  );
+  await showPage.flashMessage.clickLast();
+  await save;
+  assert.equal(
+    currentRouteName(),
+    'vault.cluster.access.identity.show',
+    `${itemType}: navigates to show on create`
+  );
+  assert.ok(showPage.nameContains(name), `${itemType}: renders the name on the show page`);
 
-  andThen(() => {
-    assert.equal(indexPage.items.filterBy('name', name).length, 0, `${itemType}: the row is deleted`);
-    indexPage.flashMessage.latestMessage.startsWith(
-      'Successfully deleted',
-      `${itemType}: shows flash message`
-    );
-  });
+  await indexPage.visit({ item_type: itemType });
+  assert.equal(
+    indexPage.items.filterBy('name', name).length,
+    1,
+    `${itemType}: lists the entity in the entity list`
+  );
+  await indexPage.items.filterBy('name', name)[0].menu();
+  await indexPage.delete();
+  let deleted = indexPage.confirmDelete();
+
+  await indexPage.flashMessage.waitForFlash();
+  indexPage.flashMessage.latestMessage.startsWith('Successfully deleted', `${itemType}: shows flash message`);
+  await indexPage.flashMessage.clickLast();
+  await deleted;
+  assert.equal(indexPage.items.filterBy('name', name).length, 0, `${itemType}: the row is deleted`);
 };
 
-export const testDeleteFromForm = (name, itemType, assert) => {
-  page.visit({ item_type: itemType });
-  page.editForm.name(name).submit();
-  showPage.edit();
-  andThen(() => {
-    assert.equal(
-      currentRouteName(),
-      'vault.cluster.access.identity.edit',
-      `${itemType}: navigates to edit on create`
-    );
-  });
-  page.editForm.delete().confirmDelete();
-  andThen(() => {
-    assert.equal(
-      currentRouteName(),
-      'vault.cluster.access.identity.index',
-      `${itemType}: navigates to list page on delete`
-    );
-    assert.equal(
-      indexPage.items.filterBy('name', name).length,
-      0,
-      `${itemType}: the row does not show in the list`
-    );
-    indexPage.flashMessage.latestMessage.startsWith(
-      'Successfully deleted',
-      `${itemType}: shows flash message`
-    );
-  });
+export const testDeleteFromForm = async (name, itemType, assert) => {
+  await page.visit({ item_type: itemType });
+
+  let save = page.editForm.name(name).submit();
+
+  await showPage.flashMessage.waitForFlash();
+  await showPage.flashMessage.clickLast();
+  await save;
+  await showPage.edit();
+  assert.equal(
+    currentRouteName(),
+    'vault.cluster.access.identity.edit',
+    `${itemType}: navigates to edit on create`
+  );
+  await page.editForm.delete();
+  let deleted = page.editForm.confirmDelete();
+  await indexPage.flashMessage.waitForFlash();
+  indexPage.flashMessage.latestMessage.startsWith('Successfully deleted', `${itemType}: shows flash message`);
```

Comment:
Is this supposed to be an assertion?

Reply:
lol ummm... probably yes  - I'll add that back

Reply:
added asserts and used withFlash helper here: https://github.com/hashicorp/vault/pull/5386/commits/db2f0a8969380cd104193defc04e775d266dd8c6